### PR TITLE
refactor(tools): separate time functions from tools.utils

### PR DESCRIPTION
**Pull request by Water-Melon**
KAG-2955

### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* [Implement ...]

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

File: kong/tools/time.lua [hunk 1]
```diff
@@ -0,0 +1,101 @@
+local ffi = require "ffi"
+
+
+local C             = ffi.C
+local ffi_new       = ffi.new
+
```
Comment (chronolaw) [inline kong/tools/time.lua:6]: add a line:
```
local tonumber      = tonumber
```

File: kong/tools/utils.lua [hunk 2]
```diff
@@ -32,19 +30,12 @@ local split         = pl_stringx.split
...
-typedef int clockid_t;
-typedef struct timespec {
-        time_t   tv_sec;        /* seconds */
-        long     tv_nsec;       /* nanoseconds */
-} nanotime;
-
-int clock_gettime(clockid_t clk_id, struct timespec *tp);
 
+ffi.cdef[[
 int gethostname(char *name, size_t len);
```
Comment (chronolaw) [inline kong/tools/utils.lua:35]: Since `gethostname` does not be used, could we remove this ffi.cdef block?

Comment (Water-Melon) [reply kong/tools/utils.lua:35]: no, it cannot be removed. It is currently being used in some other places.

File: kong/tools/time.lua (hunk 1 above)
Comment (Water-Melon) [reply kong/tools/time.lua:7]: added

File: kong/tools/utils.lua (hunk 2 above)
Comment (chronolaw) [reply kong/tools/utils.lua:35]: I found it, in `pdk/node.lua`, we can remove it in another PR.

Comment (chronolaw) [reply kong/tools/utils.lua:35]: When this PR is merged, I will do it.

File: kong/tools/time.lua [hunk 3]
```diff
@@ -0,0 +1,102 @@
+local ffi = require "ffi"
+
+
+local C             = ffi.C
+local ffi_new       = ffi.new
```
Comment (chronolaw) [inline kong/tools/time.lua:5]: Since `ffi_new` is not in hot path, I think that we can remove this line and use `ffi.new` directly.

Comment (Water-Melon) [reply kong/tools/time.lua]: removed

**Review by chronolaw — APPROVED** [review]

**Review by windmgc — APPROVED** [review]

**Pull request merged**